core/lib/services/writables.ts

```diff
@@ -7,10 +7,28 @@ export const testToNormalizeValue =
 	(value: T) =>
 		filter(value) ? value : INVALID_VALUE;
 
+const numberNormalizeFn = testToNormalizeValue(isNumber);
+
 export const typeNumber: WritableWithDefaultOptions<number> = {
-	normalizeValue: testToNormalizeValue(isNumber),
+	normalizeValue: numberNormalizeFn,
 };
 
+/**
+ * A factory function that creates a type guard function to check and rectify a value is within a specified range.
+ *
+ * @param min - The minimum value allowed.
+ * @param max - The maximum value allowed.
+ * @returns A type guard function that returns the clamp value if the value is a value number, and INVALID_VALUE otherwise.
+ */
+export function typeNumberInRangeFactory(min: number, max: number) {
```

Comment:
Quite useful new guard !
I guess we need to use it many of our current numbers (pagination collectionSize / pageSize / size, rating maxRating... etc)
Do you want to do this in this PR or create a different issue ?

Reply:
Thanks for your comments. 

For this one, yes, I would prefer to do it separately, to take the time to find all location where it can used (I changed something in the rating I think to use the util `clamp` function instead of a local one, but maybe it can be done differently). Anyway, another PR would be better.